Tasks/NpmAuthenticateV0/npmauth.ts

```diff
@@ -59,6 +62,38 @@ async function main(): Promise<void> {
         util.saveFileWithName(npmrc, npmrcTable[npmrc], saveNpmrcPath);
     }
 
+    let npmrcFile = fs.readFileSync(npmrc, 'utf8').split(os.EOL);
+
+#if WIF
+    const feedUrl = npmrcparser.NormalizeRegistry(tl.getInput("feedUrl"));
+    const entraWifServiceConnectionName = tl.getInput("workloadIdentityServiceConnection");
+
+    // Skip npmrc parsing if we are using feed url and wif service connection
+    if (feedUrl && entraWifServiceConnectionName) {
+        tl.debug(tl.loc("Info_AddingFederatedFeedAuth", entraWifServiceConnectionName, feedUrl));
+        const feedTenant = await getFeedTenantId(feedUrl);
+        let token = await getFederatedWorkloadIdentityCredentials(entraWifServiceConnectionName, feedTenant);
+        if(token)
+        {
+            const nerfed = util.toNerfDart(feedUrl);
+            const auth = `${nerfed}:_authToken=${token}`;
+            tl.setSecret(token); // Mask the secret in the log
```

Comment:
the common library also sets the secret, so I would love to know if you found this set secret call necessary in your testing. I dont mind having duplicates here though!

Reply:
Oh great, then we do not need this.